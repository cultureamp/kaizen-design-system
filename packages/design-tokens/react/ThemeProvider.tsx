import React from "react"
import { ThemeManager, defaultTheme, Theme } from "../"

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager: ThemeManager
  children: React.ReactNode
}) => {
  const [theme, setTheme] = React.useState<Theme>(
    themeManager.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme) => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManager.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManager.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <ThemeContext.Provider value={theme}>
      {props.children}
    </ThemeContext.Provider>
  )
}

export const useTheme = () => React.useContext(ThemeContext)
