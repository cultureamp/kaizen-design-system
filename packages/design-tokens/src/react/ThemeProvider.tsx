import React, { useState } from "react"
import { ThemeManager, defaultTheme, Theme } from ".."

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
/**
 * @deprecated ThemeProvider is deprecated. Please use KaizenProvider from "@kaizen/components" instead.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager?: ThemeManager
  children: React.ReactNode
}): JSX.Element => {
  const [themeManagerInstance] = useState(
    () => themeManager || new ThemeManager(defaultTheme)
  )

  const [theme, setTheme] = React.useState<Theme>(
    themeManagerInstance.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme): void => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManagerInstance.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManagerInstance.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <>
      <ThemeContext.Provider value={theme}>
        {props.children}
      </ThemeContext.Provider>
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-demi-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-medium.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-regular.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/tiempos/tiempos-headline-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
    </>
  )
}

export const useTheme = (): Theme => React.useContext(ThemeContext)
