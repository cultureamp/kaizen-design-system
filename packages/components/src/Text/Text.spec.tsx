import * as React from 'react'
import { render } from '@testing-library/react'
import { AllowedTextTags, Text, TextVariants } from '.'

describe('<Text />', () => {
  describe('changes rendered HTML element when passed tag', () => {
    const testCases: TextVariants[] = [
      'intro-lede',
      'body',
      'small',
      'extra-small',
    ]

    testCases.forEach((variant) => {
      it(`renders the correct element for <Text variant={${variant}} />`, () => {
        const textMock = render(
          <Text variant={variant} tag="div">
            Example
          </Text>,
        )
        expect(textMock.getByText('Example').tagName).toBe('DIV')
      })
    })
  })

  it('passes through data attributes', () => {
    const { getByTestId } = render(
      <Text variant="intro-lede" data-testid="test-id">
        Example
      </Text>,
    )
    expect(getByTestId('test-id')).toBeInTheDocument()
  })

  describe('defaults to the correct HTML element', () => {
    type TestObject = { variant: TextVariants; el: AllowedTextTags }
    const testCases: TestObject[] = [
      { variant: 'intro-lede', el: 'p' },
      { variant: 'body', el: 'p' },
      { variant: 'small', el: 'p' },
      { variant: 'extra-small', el: 'p' },
    ]

    testCases.forEach(({ variant, el }) => {
      it(`renders the correct element for <Text variant={${variant}} />`, () => {
        const textMock = render(<Text variant={variant}>Example</Text>)
        expect(textMock.getByText('Example').tagName.toLowerCase()).toBe(el)
      })
    })
  })
})
