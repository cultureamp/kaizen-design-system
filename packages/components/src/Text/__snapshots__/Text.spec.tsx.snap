// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Text /> > defaults to the correct HTML element > renders the correct element for <Text variant={body} /> 1`] = `
<body>
  <div>
    <p
      class="_text_cb8147 _body_cb8147 _dark_cb8147"
    >
      Example
    </p>
  </div>
</body>
`;

exports[`<Text /> > defaults to the correct HTML element > renders the correct element for <Text variant={extra-small} /> 1`] = `
<body>
  <div>
    <p
      class="_text_cb8147 _extra-small_cb8147 _dark_cb8147"
    >
      Example
    </p>
  </div>
</body>
`;

exports[`<Text /> > defaults to the correct HTML element > renders the correct element for <Text variant={intro-lede} /> 1`] = `
<body>
  <div>
    <p
      class="_text_cb8147 _intro-lede_cb8147 _dark_cb8147"
    >
      Example
    </p>
  </div>
</body>
`;

exports[`<Text /> > defaults to the correct HTML element > renders the correct element for <Text variant={small} /> 1`] = `
<body>
  <div>
    <p
      class="_text_cb8147 _small_cb8147 _dark_cb8147"
    >
      Example
    </p>
  </div>
</body>
`;

exports[`<Text /> defaults to the correct HTML element renders the correct element for <Text variant={body} /> 1`] = `
<body>
  <div>
    <p
      class="text body dark"
    >
      Example
    </p>
  </div>
</body>
`;

exports[`<Text /> defaults to the correct HTML element renders the correct element for <Text variant={extra-small} /> 1`] = `
<body>
  <div>
    <p
      class="text extra-small dark"
    >
      Example
    </p>
  </div>
</body>
`;

exports[`<Text /> defaults to the correct HTML element renders the correct element for <Text variant={intro-lede} /> 1`] = `
<body>
  <div>
    <p
      class="text intro-lede dark"
    >
      Example
    </p>
  </div>
</body>
`;

exports[`<Text /> defaults to the correct HTML element renders the correct element for <Text variant={small} /> 1`] = `
<body>
  <div>
    <p
      class="text small dark"
    >
      Example
    </p>
  </div>
</body>
`;
